src/io.cpp

```diff
@@ -97,16 +97,24 @@ const Catalog &CatalogRead() {
         char *tsvPath = getenv("LOST_BSC_PATH");
         catalog = BscParse(tsvPath ? tsvPath : DEFAULT_BSC_PATH);
         // perform essential narrowing
-        // remove all stars with exactly the same position as another, keeping the one with brighter magnitude
+        // remove all stars with exactly the same position as another, keeping the one with brighter
+        // magnitude
         std::sort(catalog.begin(), catalog.end(), [](const CatalogStar &a, const CatalogStar &b) {
-            return a.spatial.x < b.spatial.x;
+            if (a.spatial.x != b.spatial.x) {
+                return a.spatial.x < b.spatial.x;
+            } else if (a.spatial.y != b.spatial.y) {
+                return a.spatial.y < b.spatial.y;
+            }
+            return a.spatial.z < b.spatial.z;
         });
-        for (int i = catalog.size(); i > 0; i--) {
-            if ((catalog[i].spatial - catalog[i-1].spatial).Magnitude() < 5e-5) { // 70 stars removed at this threshold.
-                if (catalog[i].magnitude > catalog[i-1].magnitude) {
-                    catalog.erase(catalog.begin() + i);
-                } else {
+        for (int i = catalog.size() - 1; i > 0; i--) {  // [BUG]? catalog[catalog.size()] is invalid
+            if ((catalog[i].spatial - catalog[i - 1].spatial).Magnitude() < 5e-5) {  // 70 stars removed at this threshold.
```

Comment:
Yep, this loop indexing looks like a bug, good catch!

(Valgrind didn't find this error because the catalog is just part of the MultiDatabase -- so the first iteration of the loop read from the wrong part of the MultiDatabase, which is still memory allocated to the process).